packages/store/ts/codegen/renderTable.ts

```diff
@@ -80,27 +78,14 @@ export function renderTable(options: RenderTableOptions) {
   
       ${renderFieldLayout(fields)}
 
-      /** 
-       * @notice Get the table's key schema.
-       * @return _keySchema The key schema for the table.
-       */
-      function getKeySchema() internal pure returns (Schema) {
-        SchemaType[] memory _keySchema = new SchemaType[](${keyTuple.length});
-        ${renderList(keyTuple, ({ enumName }, index) => `_keySchema[${index}] = SchemaType.${enumName};`)}
-
-        return SchemaLib.encode(_keySchema);
-      }
-
-      /**
-       * @notice Get the table's value schema.
-       * @return _valueSchema The value schema for the table.
-       */
-      function getValueSchema() internal pure returns (Schema) {
-        SchemaType[] memory _valueSchema = new SchemaType[](${fields.length});
-        ${renderList(fields, ({ enumName }, index) => `_valueSchema[${index}] = SchemaType.${enumName};`)}
-
-        return SchemaLib.encode(_valueSchema);
-      }
+      // Hex-encoded key schema of (${keyTuple.map((field) => field.internalTypeId).join(", ")})
+      Schema constant _keySchema = Schema.wrap(${keySchemaToHex(
+        Object.fromEntries(keyTuple.map((field) => [field.name, field.internalTypeId])) as KeySchema
+      )});
+      // Hex-encoded value schema of (${fields.map((field) => field.internalTypeId).join(", ")})
+      Schema constant _valueSchema = Schema.wrap(${valueSchemaToHex(
+        Object.fromEntries(fields.map((field) => [field.name, field.internalTypeId])) as ValueSchema
+      )});
```

Comment:
I generally agree with this change but am a little wary of this being a change to an audited code path. Are there any tests we can add to really make sure the typescript encoding matches the expected encoding? maybe a couple forge test cases where we encode with the new method and confirm it matches what would be produced by `SchemaLib.encode`?

Reply:
how about this? https://github.com/latticexyz/mud/pull/2328/commits/39fa54244aa141416953e6a32e698a34aa7854a9

Reply:
Perfect!